twitter/twitter_client.bal

```diff
@@ -0,0 +1,1057 @@
+    # + maxId - Returns results with an ID less than (that is, older than) or equal to the specified ID.
+    # + trimUser - When set to true, each Tweet returned in a timeline will include a user object including only the
+    #              status authors numerical ID. Omit this parameter to receive the complete user object.
+    # + excludeReplies - This parameter will prevent replies from appearing in the returned timeline.
+    # + includeEntities - The entities node will not be included when set to false.
+    # + return - Returns a collection of Tweet object when successful, else returns an error.
+    public remote function getHomeTimelineTweets(int? count = (), int? sinceId = (), int? maxId = (),
+                                                 boolean? trimUser = (), boolean? excludeReplies = (),
+                                                 boolean? includeEntities = ()) returns Tweet[]|error;
+
+    # Returns the 20 most recent mentions (Tweets containing a users's @screen_name) for the authenticating user.
+    # The timeline returned is the equivalent of the one seen when you view your mentions on twitter.com.
+    #
+    # + count - Specifies the number of Tweets to try and retrieve, up to a maximum of 200.
+    # + sinceId - Returns results with an ID greater than (that is, more recent than) the specified ID.
+    # + maxId - Returns results with an ID less than (that is, older than) or equal to the specified ID.
+    # + trimUser - When set to true, each tweet returned in a timeline will include a user object including only
+    #              the status authors numerical ID. Omit this parameter to receive the complete user object.
+    # + includeEntities - The entities node will not be included when set to false.
+    # + return - Returns a collection of Tweet object when successful, else returns an error.
+    public remote function getTweetMentions(int? count = (), int? sinceId = (), int? maxId = (), boolean? trimUser = (),
+                                            boolean? includeEntities = ()) returns Tweet[]|error;
+};
+
+# Twitter User Client object.
+#
+# + httpClient - HTTP Client.
+# + twitterClient - Twitter Client.
+public type TwitterUserClient client object {
+    http:Client httpClient;
+    TwitterClient|error twitterClient;
+
+    public function __init(TwitterClient|TwitterConfiguration config) returns error? {
+        if (config is TwitterClient) {
+            self.httpClient = config.getHttpClient();
+            self.twitterClient = config;
+        } else {
+            self.httpClient = new(TWITTER_API_URL, config = config.clientConfig);
+            var isValid = verifyCredentials(self.httpClient, config.clientId, config.clientSecret, config.accessToken,
+                config.accessTokenSecret);
+            if (isValid is error) {
+                return isValid;
+            }
+            self.twitterClient = new(config);
```

Comment:
We can replace L191 - L197 as follows right?
```ballerina
            self.twitterClient = check new(config); // assuming we change the variable type
            self.httpClient = self.twitterClient.getHttpClient();
```

With the current impl we are unnecessarily creating two `http:Client`s and also calling `verifyCredentials` twice right?

Reply:
Same applies for TwitterTrendsClient similarly.

Reply:
I changed the variable type like
TwitterClient twitterClient;

self.twitterClient = check new(config);

it gives error as "invalid usage of the 'check' expression operator: no expression type is equivalent to error type"

Reply:
Can you try changing this as follows:
```ballerina
     self.twitterClient = check new TwitterClient(config);
```

@pubudu91 shouldn't this work even without explicitly specifying the type with new?

Reply:
This syntax is working fine